src/python/pants/backend/shell/lint/shfmt/subsystem.py

```diff
@@ -18,12 +18,16 @@ class Shfmt(TemplatedExternalTool):
     name = "shfmt"
     help = "An autoformatter for shell scripts (https://github.com/mvdan/sh)."
 
-    default_version = "v3.2.4"
+    default_version = "v3.6.0"
     default_known_versions = [
         "v3.2.4|macos_arm64 |e70fc42e69debe3e400347d4f918630cdf4bf2537277d672bbc43490387508ec|2998546",
         "v3.2.4|macos_x86_64|43a0461a1b54070ddc04fbbf1b78f7861ee39a65a61f5466d15a39c4aba4f917|2980208",
         "v3.2.4|linux_arm64 |6474d9cc08a1c9fe2ef4be7a004951998e3067d46cf55a011ddd5ff7bfab3de6|2752512",
         "v3.2.4|linux_x86_64|3f5a47f8fec27fae3e06d611559a2063f5d27e4b9501171dde9959b8c60a3538|2797568",
```

Comment:
I'm thinking it'd be good for us to start keeping the old default versions around in this option -- it makes it easier for users to simply change `[tool].version` to keep the same version as a prior Pants release.

The only downsides are these 4 extra lines in our code + longer reference docs because the option's default value will be longer. Seems reasonable.

cc @jriddy - thoughts?

Reply:
Makes sense to me. We can special-case this in the reference docs so we show just the first few lines and then an ellipsis.